### PR TITLE
Add new labels to Services

### LINKED ISSUE #1405: Add kubernetes labels to help with selectors
Add new kubernetes labels to be able to select pods/deployments/services by e.g. name/canary/prod/staging

### PR DESCRIPTION
**Pull request by glindsell**
Fixes #1405

### CODE REVIEW

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **glindsell**

If they are not already assigned, you can assign the PR to them by writing `/assign @glindsell` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by seldondev** [comment]
Tue Mar 17 16:41:12 UTC 2020
The logs for [lint] [2] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1555/2.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1555 --build=2

**Comment by seldondev** [comment]
Tue Mar 17 16:42:11 UTC 2020
The logs for [pr-build] [1] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1555/1.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1555 --build=1

File: operator/apis/machinelearning/v1/seldondeployment_types.go [hunk 1]
```diff
@@ -185,8 +196,9 @@ type PredictorSpec struct {
 	Traffic         int32                   `json:"traffic,omitempty" protobuf:"bytes,9,opt,name=traffic"`
 	Explainer       *Explainer              `json:"explainer,omitempty" protobuf:"bytes,10,opt,name=explainer"`
 	Shadow          bool                    `json:"shadow,omitempty" protobuf:"bytes,11,opt,name=shadow"`
-	Protocol        Protocol                `json:"protocol,omitempty" protobuf:"bytes,12,opt,name=protocol"`
-	Transport       Transport               `json:"transport,omitempty" protobuf:"bytes,13,opt,name=transport"`
+	Default         bool                    `json:"default,omitempty" protobuf:"bytes,12,opt,name=default"`
```
Comment (cliveseldon) [inline operator/apis/machinelearning/v1/seldondeployment_types.go]: Do we need to change the CRD to add default? As it really not anything that changes functionality

Comment (glindsell) [reply operator/apis/machinelearning/v1/seldondeployment_types.go]: I thought that might be a contentious addition. Happy to remove it if there is no use for it. Could I add logic to say that if a service is not a shadow it automatically gets labelled as a default instead in that case?

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **cliveseldon**

If they are not already assigned, you can assign the PR to them by writing `/assign @cliveseldon` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by seldondev** [comment]
@glindsell: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pr-build | accdda691d8cc1e251370d0add97fe6e8948f392 | [link]() | `/test this`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
@glindsell: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pr-build | accdda691d8cc1e251370d0add97fe6e8948f392 | [link]() | `/test this`
lint | accdda691d8cc1e251370d0add97fe6e8948f392 | [link]() | `/test lint`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

File: operator/apis/machinelearning/v1/seldondeployment_types.go (hunk 1 above)
Comment (glindsell) [reply operator/apis/machinelearning/v1/seldondeployment_types.go]: Have replaced the Default with Canary, which is a more useful flag

**Comment by seldondev** [comment]
@glindsell: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pr-build | accdda691d8cc1e251370d0add97fe6e8948f392 | [link]() | `/test this`
lint | accdda691d8cc1e251370d0add97fe6e8948f392 | [link]() | `/test lint`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
@glindsell: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
lint | accdda691d8cc1e251370d0add97fe6e8948f392 | [link]() | `/test lint`
pr-build | accdda691d8cc1e251370d0add97fe6e8948f392 | [link]() | `/test this`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
Wed Mar 18 12:44:35 UTC 2020
The logs for [lint] [8] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1555/8.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1555 --build=8

**Comment by seldondev** [comment]
Wed Mar 18 12:45:35 UTC 2020
The logs for [pr-build] [7] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1555/7.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1555 --build=7

File: operator/apis/machinelearning/v1/seldondeployment_types.go (hunk 1 above)
Comment (glindsell) [reply operator/apis/machinelearning/v1/seldondeployment_types.go]: Removed new field entirely now and replaced with logic based on Traffic %

File: operator/controllers/seldondeployment_controller.go [hunk 2]
```diff
@@ -679,6 +680,30 @@ func createContainerService(deploy *appsv1.Deployment, p machinelearningv1.Predi
 			SessionAffinity: corev1.ServiceAffinityNone,
 		},
 	}
+	switch *pu.Type {
+	case machinelearningv1.ROUTER:
```
Comment (cliveseldon) [inline operator/controllers/seldondeployment_controller.go]: Can we move to a separate file with some tests?

**Comment by seldondev** [comment]
Mon Mar 23 15:04:07 UTC 2020
The logs for [lint] [10] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1555/10.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1555 --build=10

File: operator/controllers/seldondeployment_controller.go (hunk 2 above)
Comment (glindsell) [reply operator/controllers/seldondeployment_controller.go]: @cliveseldon Fixed

**Comment by seldondev** [comment]
Mon Mar 23 15:04:22 UTC 2020
The logs for [pr-build] [9] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1555/9.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1555 --build=9

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/SeldonIO/seldon-core/pull/1555#issuecomment-603299094" title="Approved">cliveseldon</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)~~ [cliveseldon]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by seldondev** [comment]
Tue Mar 24 15:19:02 UTC 2020
The logs for [pr-build] [11] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1555/11.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1555 --build=11

**Comment by seldondev** [comment]
Tue Mar 24 15:19:02 UTC 2020
The logs for [lint] [12] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-1555/12.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-1555 --build=12

**Pull request merged**